slang-gfx.h

```diff
@@ -1183,6 +1178,266 @@ struct WindowHandle
     }
 };
 
+enum class ResourceState
+{
+    Undefined,
+    ShaderResource,
+    UnorderedAccess,
+    RenderTarget,
+    DepthRead,
+    DepthWrite,
+    Present,
+    CopySource,
+    CopyDestination,
+    ResolveSource,
+    ResolveDestination,
+};
+
+struct FaceMask
+{
+    enum Enum
+    {
+        Front = 1, Back = 2
+    };
+};
+
+class IRenderPassLayout : public ISlangUnknown
+{
+public:
+    enum class AttachmentLoadOp
+    {
+        Load, Clear, DontCare
+    };
+    enum class AttachmentStoreOp
+    {
+        Store, DontCare
+    };
+    struct AttachmentAccessDesc
+    {
+        AttachmentLoadOp loadOp;
+        AttachmentLoadOp stencilLoadOp;
+        AttachmentStoreOp storeOp;
+        AttachmentStoreOp stencilStoreOp;
+        ResourceState initialState;
+        ResourceState finalState;
+    };
+    struct Desc
+    {
+        IFramebufferLayout* framebufferLayout;
+        uint32_t renderTargetCount;
+        AttachmentAccessDesc* renderTargetAccess;
+        AttachmentAccessDesc* depthStencilAccess;
+    };
+};
+#define SLANG_UUID_IRenderPassLayout                                                   \
+    {                                                                                  \
+        0xdaab0b1a, 0xf45d, 0x4ae9, { 0xbf, 0x2c, 0xe0, 0xbb, 0x76, 0x7d, 0xfa, 0xd1 } \
+    }
+
+class ICommandEncoder : public ISlangUnknown
```

Comment:
Because the validity of encoders isn't really coupled to reference counts, should we make them not use `ISlangUnknown` and require `ComPtr` or similar? We could conceivably make them be non-reference-counted objects that just "magically" go away on `endEncoding`.

I don't know whether that would actually be wise, or if keep the reference-counting around is a net win.